assignment1.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"fmt"
+	"math"
+)
+
+func hashGenerator(str string) uint64 {
+	var base, hash uint64
+
+	base = 257
+	bucketSize = uint64(math.Pow(2, 64))
+
+	hash = 0
+	for i := 0; i < len(str); i++ {
+		ascii := uint64(str[i])
+		
+		hash = (hash * base + ascii) % bucketSize
```

Comment:
In 64-bit arithmetic, overflow happens naturally; multiplying and adding uint64 values are already performed modulo 2^64. This makes the explicit modulo operation redundant if you intend to work with all 64 bits
```suggestion
		hash = hash * base + ascii
```